packages/expo-calendar/ios/CalendarModule.swift

```diff
@@ -0,0 +1,509 @@
+      try checkCalendarPermissions()
+      let calendarEvent = try getCalendar(from: event)
+      let span: EKSpan = options.futureEvents == true ? .futureEvents : .thisEvent
+      calendarEvent.title = event.title
+      calendarEvent.location = event.location
+      calendarEvent.notes = event.notes
+
+      if let timeZone = event.timeZone {
+        if let tz = TimeZone(identifier: timeZone) {
+          calendarEvent.timeZone = tz
+        } else {
+          throw InvalidTimeZoneException()
+        }
+      }
+
+      calendarEvent.alarms = createCalendarEventAlarms(alarms: event.alarms)
+      if let rule = event.recurrenceRule {
+        let newRule = createRecurrenceRule(rule: rule)
+        if let newRule {
+          calendarEvent.recurrenceRules = [newRule]
+        }
+      }
+
+      if let urlString = event.url?.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed), let url = URL(string: urlString) {
+        calendarEvent.url = url
+      }
+
+      if let startDate = event.startDate {
+        calendarEvent.startDate = parse(date: startDate)
+      }
+      if let endDate = event.startDate {
+        calendarEvent.endDate = parse(date: endDate)
+      }
+
+      calendarEvent.isAllDay = event.allDay
+      calendarEvent.availability = getAvailability(availability: event.availability)
+      try eventStore.save(calendarEvent, span: span, commit: true)
+      return calendarEvent.calendarItemIdentifier
+    }
+
+    AsyncFunction("deleteEventAsync") { (event: Event, options: RecurringEventOptions) in
+      try checkCalendarPermissions()
+      guard let id = event.id else {
+        throw EventIdRequiredException()
+      }
+      let span: EKSpan = options.futureEvents == true ? .futureEvents : .thisEvent
+
+      let instanceStartDate = parse(date: event.instanceStartDate)
+      let calendarEvent = getEvent(with: id, startDate: instanceStartDate)
+
+      guard let calendarEvent else {
+        return
+      }
+      try eventStore.remove(calendarEvent, span: span)
+    }
+
+    AsyncFunction("getAttendeesForEventAsync") { (event: Event) -> [[String: Any?]] in
+      try checkCalendarPermissions()
+      guard let id = event.id else {
+        throw EventIdRequiredException()
+      }
+      let instanceStartDate = parse(date: event.instanceStartDate)
+
+      let item = getEvent(with: id, startDate: instanceStartDate)
+
+      guard let item, let attendees = item.attendees else {
+        return []
+      }
+
+      return serialize(attendees: attendees)
+    }
+
+    AsyncFunction("getRemindersAsync") { (startDateStr: String, endDateStr: String, calendarIds: [String], status: String?, promise: Promise) in
+      try checkRemindersPermissions()
+      var reminderCalendars = [EKCalendar]()
+      let startDate = parse(date: startDateStr)
+      let endDate = parse(date: endDateStr)
+
+      if calendarIds.isEmpty {
+        promise.reject(CalendarIdRequiredException())
```

Comment:
Do we need to require calendarIds? Why not just search all calendars if no ids are provided?

Reply:
Yeah, we can do that. The TS types accept null so it shouldn't be required 👍